.buildkite/pipeline.yml

```diff
@@ -1,3 +1,82 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0'
+  DOCKER_COMPOSE_VERSION: '1.25.5'
+  DOCKER_REGISTRY: "docker.elastic.co"
+  DOCKER_IMAGE: "${DOCKER_REGISTRY}/observability-ci/temp-fleet-server" # needs to rename after tests
+  DOCKER_IMAGE_TAG: ""
+  DOCKER_IMAGE_SHA_TAG: "temp-${BUILDKITE_COMMIT}" # needs to rename after tests
+  DOCKER_IMAGE_LATEST_TAG: "temp-latest" # needs to rename after tests
+  DOCKER_IMAGE_GIT_TAG: "temp-${BUILDKITE_BRANCH}" # needs to rename after tests
+  GO_AGENT_IMAGE: "golang:1.20.3"
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Check and build"
+    key: "check"
+    steps:
+      - label: ":go: Run check-ci"
+        key: check-ci
+        command: "make check-ci"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+      - label: ":go: Run local build"
+        key: make-local
+        command: "make local"
+        agents:
+          image: "${GO_AGENT_IMAGE}"
+          cpu: "8"
+          memory: "4G"
+
+  - group: "Run tests"
+    key: "tests"
+    steps:
+      - label: "Run unit tests"
+        key: unit-test
+        commands:
+          - ".buildkite/scripts/setenv.sh with-go"
+          - "make test-unit junit-report"
+        agents:
+          provider: "gcp"
+        artifact_paths:
+          - build/*.xml
+
+      - label: "Run integration tests"
+        key: int-test
+        commands:
+          - ".buildkite/scripts/setenv.sh with-go-docker-compose"
+          - "make test-int junit-report"
+        agents:
+          provider: "gcp"
+        artifact_paths:
+          - build/*.xml
+
+      - label: ":junit: Junit annotate"
+        plugins:
+          - junit-annotate#v2.4.1:
+              artifacts: "build/*.xml"
+        agents:
+          provider: "gcp"
+        depends_on:
+          - step: "unit-test"
+            allow_failure: true
+          - step: "int-test"
+            allow_failure: true
+    depends_on: "check"
+
+  - group: "Publish docker image"
```

Comment:
This step should be executed following the existing conditions from the [Jenkinsfile](https://github.com/elastic/fleet-server/blob/e7d04c9d6eb40e04b679dc8f99cc5d7f00ec553a/.ci/Jenkinsfile#LL136C4-L138C8)